### PR TITLE
Add learning rate and weight decay arguments to PyTorch estimators

### LINKED ISSUE #2288: Learning rate setting not available for Pytorch models
Hello everyone,

I hesitated to tag this issue as a "bug" because I think it's a simple design problem.

Indeed, unlike the **MXNet** equivalent, in the **Pytorch** estimators, the trainer does not carry the `learning_rate` information. 
It's defined in LightningModule, but this one is not instantiated before training: 

```
def train_model(
    self,
    training_data: Dataset,
    validation_data: Optional[Dataset] = None,
    num_workers: int = 0,
    shuffle_buffer_length: Optional[int] = None,
    cache_data: bool = False,
    ckpt_path: Optional[str] = None,
    **kwargs,
) -> TrainOutput:transformation = self.create_transformation()

    transformed_training_data = transformation.apply(
        training_data, is_train=True
    )
    
    training_network = self.create_lightning_module()  <== here
    
    ...
    
    trainer.fit(
            model=training_network,
            train_dataloaders=training_data_loader,
            val_dataloaders=validation_data_loader,
            ckpt_path=ckpt_path,
        )
    
    logger.info(f"Loading best model from {checkpoint.best_model_path}")
    best_model = training_network.load_from_checkpoint(
        checkpoint.best_model_path
    )
    
    return TrainOutput(
        transformation=transformation,
        trained_net=best_model,
        trainer=trainer,
        predictor=self.create_predictor(transformation, best_model),
    )
```

So I have the impression that it's not possible to set the learning rate at the estimator initialization, nor to specify it later during the `train` call.

Am I missing something?

Thanks in advance !

### PR DESCRIPTION
**Pull request by shubhamkapoor**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Review by lostella — APPROVED** [review]
Looks good to me, thanks @shubhamkapoor 👍

File: src/gluonts/torch/model/deepar/estimator.py [hunk 1]
```diff
@@ -95,6 +95,10 @@ class DeepAREstimator(PyTorchLightningEstimator):
         Number of RNN layers (default: 2).
     hidden_size
         Number of RNN cells for each layer (default: 40).
+    lr
```
Comment (jaheba) [inline src/gluonts/torch/model/deepar/estimator.py:98]: Can we use `learning_rate` to be consistent with our `mx` version?

**Review by jaheba — CHANGES_REQUESTED** [review]
Let's check consistency before merging.

**Comment by lostella** [comment]
> Let's check consistency before merging.

The thing is, `lr` is consistent with the PyTorch ecosystem (see https://pytorch.org/docs/stable/optim.html) which would also be desirable in my opinion

File: src/gluonts/torch/model/deepar/estimator.py (hunk 1 above)
Comment (shubhamkapoor) [reply src/gluonts/torch/model/deepar/estimator.py:98]: `mx` version doesn't use `learning_rate` directly in estimator, but rather in `mx` trainer. So as @lostella mentioned, I think `lr` is consistent with `torch` trainer/optimizer. And to make `mx` and `torch` interfaces consistent, we can use `lr` in `mx` trainer as well.

**Review by jaheba — APPROVED** [review]
I guess we have to die at least one death :)
